### PR TITLE
Add Tests for branch, RV32A, RV32F and RV32F extensions.

### PR DESCRIPTION
**Pull request by faisalill**
### What's changed?
* Added tests for branch, atomic, F and D extension instructions.
* Created a seperate section for multiplication tests and moved 
M extension tests there.

### CODE REVIEW

**Review by skudlur — COMMENTED** [review]
Please add the instructions too, I will merge after all the tests pass

**Comment by faisalill** [comment]
@skudlur 
### What's Changed? 
* Added instructions for branch, RV32F, RV32A and RV32D extension.
* Added Instrucion breakdown for the same.
* Added ci/cd tests (triggers for pr and push actions)

File: src/lib.rs [hunk 1]
```diff
@@ -629,6 +809,890 @@ pub fn instruction_decoder(instr: Vec<String>) -> String {
...
+        //     let rs1_slice_joined = rs1_slice.join("");
+        //     let rd_slice = &instr[20..25];
+        //     let rd_slice_joined = rd_slice.join("");
+
+        //     let mut rd_bits = i32::from_str_radix(&rd_slice_joined, 2).unwrap();
+        //     let rs1_bits = i32::from_str_radix(&rs1_slice_joined, 2).unwrap();
+        //     let imm_bits = i32::from_str_radix(&imm_slice_joined, 2).unwrap();
+
+            
+        // }
```
Comment (faisalill) [inline src/lib.rs]: @skudlur can you remove this please...

**Review by skudlur — APPROVED** [review]
LGTM. Good job @faisalill

**Pull request merged**